lsp/src/Language/LSP/Server/Core.hs

```diff
@@ -138,6 +140,10 @@ data LanguageContextEnv config = LanguageContextEnv
     resState :: !(LanguageContextState config)
   , resClientCapabilities :: !L.ClientCapabilities
   , resRootPath :: !(Maybe FilePath)
+  , resProgressStartDelay :: Int
```

Comment:
Maybe this can be set for each progress instead of globally? Since we might want different delay for different progress

Reply:
I can't actually think of a use case for this. I think it _should_ be pretty generic?

Reply:
I'm inclined to leave it as global for now unless we have a need for it to be different.